pkg/reduce/pbq/pbq.go

```diff
@@ -68,11 +68,13 @@ func (p *PBQ) Write(ctx context.Context, request *window.TimedWindowRequest, per
 	}
 
 	// write the request to the output channel
+	// since it is a blocking write, we should have a select with context,
 	select {
 	case p.output <- request:
-
 	case <-ctx.Done():
-		return ctx.Err()
+		// we can persist the message even if the context is done that way we will not rely on
```

Comment:
How will we guarantee that the message will be written to disk?

Reply:
Writing to disk is a blocking call, and we only close the WAL once we're done processing all the messages in the batch.

Reply:
Just like No-Ack, WAL writes also don't depend on the ctx.

Reply:
Please open an issue to track